test/groovy/NeoDeployTest.groovy

```diff
@@ -122,6 +132,107 @@ class NeoDeployTest extends BasePiperTest {
+        stepRule.step.neoDeploy(
+                script: nullScript,
+                source: archiveName,
+                mtaExtensionDescriptors: 'myExtension.yml'
+        )
+
+        assert checkedExtensionFiles.contains('myExtension.yml')
+
+        assertThat(shellRule.shell,
+            new CommandLineMatcher()
+                .hasProlog('neo.sh deploy-mta')
+                .hasSingleQuotedOption('extensions', 'myExtension.yml'))
+    }
+
+    @Test
+    void extensionsAsSetTest() {
+        Set extensions= ['myExtension1.yml' ,'myExtension2.yml']
+        extensionsAsCollectionTest(extensions)
+    }
+
+    @Test
+    void extensionsAsListTest() {
+        List extensions= ['myExtension1.yml' ,'myExtension2.yml']
+        extensionsAsCollectionTest(extensions)
+    }
+    @Test
+    void sameExtensionProvidedTwiceTest() {
+        List extensions= ['myExtension1.yml' ,'myExtension2.yml', 'myExtension1.yml']
+        extensionsAsCollectionTest(extensions)
+    }
+
+    void extensionsAsCollectionTest(def extensions) {
+
+        def checkedExtensionFiles = []
+
+        StepAssertions.metaClass.static.assertFileExists =
+            { Script step, String filePath ->
+                checkedExtensionFiles << filePath
+                if( ! [
+                    archiveName,
+                    'myExtension1.yml',
+                    'myExtension2.yml',
+                    ].contains(filePath) )
+                    step.error("File ${filePath} cannot be found.")
+            }
+
+        stepRule.step.neoDeploy(
+                script: nullScript,
+                source: archiveName,
+                mtaExtensionDescriptors: extensions
+        )
+
+        assert checkedExtensionFiles.contains('myExtension1.yml')
+        assert checkedExtensionFiles.contains('myExtension2.yml')
+
+        assertThat(shellRule.shell,
+            new CommandLineMatcher()
+                .hasProlog('neo.sh deploy-mta')
+                // some kind of creative usage for the single quotation check (... single quotes inside)
+                .hasSingleQuotedOption('extensions', 'myExtension1.yml\',\'myExtension2.yml'))
+
+    }
+
+    @Test
+    void extensionsForWrongDeployModeTest() {
+
+        thrown.expect(AbortException)
+        thrown.expectMessage('Extensions are only supported for deploy mode \'MTA\'')
+
+        StepAssertions.metaClass.static.assertFileExists =
+            { Script step, String filePath ->
+              if( ! [archiveName, 'myExtension.yml'].contains(filePath) )
+                  step.error("File ${filePath} cannot be found.")
+            }
+
+        stepRule.step.neoDeploy(
+            script: nullScript,
+            source: archiveName,
+            deployMode: 'warParams',
+            mtaExtensionDescriptors: 'myExtension.yml',
+            neo:
+                [
+                    application: 'does',
+                    runtime: 'not',
+                    runtimeVersion: 'matter'
+                ],
```

Comment:
The , is not needed

Reply:
Syntactically the `,` is not needed. But it helps keeping the change small in the future. In case something is added to the map below the comma needs to be added and with the the `]` line is part of the diff. I would like to suggest to keep it for the sake of the smaller diff.

Reply:
> but then we would leave commas everywhere, 

I disagree.

I guess we have simply different mind sets, and that is OK like this.

I think it is a phantom to have the coding fully unified in each and every way. Even automated formatters (and this here is not primary a formatting issue) can ensure fully unified code. Basically we are all humans with different styles and habit and we should be open for that. I personally think we should accept that not everything looks the same.

I mean there are the edge cases like full-freedom and everything-regulated. In my opinion having "request changes" because of commas (which can be explained by shorter diff) is in my opinion a little bit too close on one of these edge points.

Nevertheless I removed the `,`s.

Reply:
ok, thank you.